P2: Add P2 branding

mindgraffiti opened the pull request:
Closes #15264

This PR exchanges the default site icon for a default P2 icon when the site is identified as `is_wpforteams_site` and has no site icon uploaded. Examples:

| Default site icon | Default P2 icon |
| --- | --- |
| <img width="47" alt="Screen Shot 2020-12-03 at 9 18 34 AM" src="https://user-images.githubusercontent.com/1062444/101048761-86c66b00-3548-11eb-8891-7f3d44deac26.png"> | <img width="80" alt="Screen Shot 2020-12-03 at 9 19 44 AM" src="https://user-images.githubusercontent.com/1062444/101048967-b1182880-3548-11eb-8259-6b0770421a83.png"> |

I **swear** this PR is not 1k+ LOC changes. 984 LOC added are from duplicating the Core Data data model file. I had to add an attribute to an existing entity 😭 

### Design Review
The Reader Blog Details `avatarImageView` (where the blavatar sits) does not come with a border or styling. So I added a 1 point border width and `listIcon` border color when the default P2 site icon displays. Not sure that's the color or border thickness we want to go with. The un-styled, original image view can be seen in the very last row of the screenshots table.

### To Test / Screenshots

| Light | Dark |
| --- | --- |
| Site Picker... |
| ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 41 02](https://user-images.githubusercontent.com/1062444/101097389-b3997300-3586-11eb-8f52-53d8ec002023.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 41 49](https://user-images.githubusercontent.com/1062444/101097408-b8f6bd80-3586-11eb-878f-b06d8c601cda.png) |
| My Sites... |
| ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 47 44](https://user-images.githubusercontent.com/1062444/101097817-81d4dc00-3587-11eb-8be3-8451675e94f6.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 47 33](https://user-images.githubusercontent.com/1062444/101097825-84cfcc80-3587-11eb-8711-bee1f866402e.png) |
| Reader Blog Details... |
| ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 41 25](https://user-images.githubusercontent.com/1062444/101097440-c613ac80-3586-11eb-8bc4-5083772e5781.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 41 42](https://user-images.githubusercontent.com/1062444/101097448-c90e9d00-3586-11eb-963e-e3ab5f81afc2.png) |
| Reader Blog Details, no border |
| ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 02 08](https://user-images.githubusercontent.com/1062444/101095160-243e9080-3583-11eb-97ff-1c78f5de0ceb.png) | ![Simulator Screen Shot - iPhone 11 Pro - 2020-12-03 at 16 02 42](https://user-images.githubusercontent.com/1062444/101095171-2acd0800-3583-11eb-969a-4e4a658c9304.png) |


PR submission checklist:

- [x] I have considered adding unit tests where possible.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

mindgraffiti committed:
Update podfile to latest Gridicons version

mindgraffiti committed:
Use constants instead of magic numbers

mindgraffiti committed:
Merge branch 'develop' into feature/15264-blavatar-p2-logo

mindgraffiti committed:
Update checksum after merge

mindgraffiti committed:
Apply site icon changes to both classes

So that the Big Titles project can continue smoothly

mindgraffiti committed:
Create a new Core Data model, select it, add attribute

mindgraffiti committed:
Point podfile to feature branch of WordPressKit

mindgraffiti committed:
Delete file reference from project file

mindgraffiti committed:
Add isWPForTeams property to ReaderTopicService

mindgraffiti committed:
Check if ReaderSiteTopic isWPForTeams type and use new default P2 icon

mindgraffiti committed:
Force podfile to refresh WPKit by pointing to a commit

mindgraffiti committed:
Add a style to the avatarImageView when displaying a P2 logo

mindgraffiti committed:
Merge branch 'develop' into feature/15264-blavatar-p2-logo

mindgraffiti committed:
Update release notes

mattmiklic commented:
It looks like we're adding a border to default icons, but only in dark mode. Is that right? I'm wondering if we can either lose the border or set it to something like the separator color so it works in both light and dark mode.

mindgraffiti committed:
Update default site icon image views to `divider` for border color

mindgraffiti commented:
@mattmiklic in the Site Picker we add a white border to default site icons at all times. I think it may have been missed when setting light / dark mode compatible colors. I've updated all 3 areas to use the `divider` color for borders and updated the screenshots with the latest.

ScoutHarris commented on WordPress/Classes/ViewRelated/Blog/WPStyleGuide+Blog.swift:
```diff
@@ -11,7 +11,7 @@ extension WPStyleGuide {
 
-        cell.imageView?.layer.borderColor = UIColor.white.cgColor
+        cell.imageView?.layer.borderColor = UIColor.divider.cgColor
         cell.imageView?.layer.borderWidth = 1
```
I think this `borderWidth` needs to be `hairlineBorderWidth` (so it adjusts with screen size). It looks awfully thick on an iPad.

<img width="194" alt="Screen Shot 2020-12-03 at 5 11 21 PM" src="https://user-images.githubusercontent.com/1816888/101105356-9797d080-358a-11eb-8b82-032e70fc3c5b.png">

ScoutHarris commented on WordPress/Classes/Extensions/UIImageView+SiteIcon.swift:
```diff
@@ -108,6 +109,14 @@ extension UIImageView {
     func downloadSiteIcon(for blog: Blog, placeholderImage: UIImage? = .siteIconPlaceholder) {
         guard let siteIconPath = blog.icon, let siteIconURL = optimizedURL(for: siteIconPath) else {
+
+            let standardSize = CGSize(width: SiteIconDefaults.imageSize, height: SiteIconDefaults.imageSize)
```
Nitpick: since `standardSize` is only used inside the following `if`, maybe move it in there? Or just nix it altogether and put the `CGSize` in the `UIImage` creation.

`image = UIImage.gridicon(.p2, size: CGSize(width: SiteIconDefaults.imageSize, height: SiteIconDefaults.imageSize))`

ScoutHarris commented:
I think the icon in the site list on the Login Epilogue need some love. At least, I don't think it's supposed to be blue.

<img width="486" alt="Screen Shot 2020-12-03 at 5 20 27 PM" src="https://user-images.githubusercontent.com/1816888/101105925-0d506c00-358c-11eb-96ff-d530fe289de0.png">

mattmiklic commented:
It looks good now with the `divider` color. Agreed with @ScoutHarris about using the hairline width. Otherwise it looks good to go!

mindgraffiti commented:
Our core data model on `develop` has significant changes (fixes to Core Data transformer warnings! 🎉 )

I will close this PR and open a series of 3 separate ones so that I can merge in my core data changes carefully.
